lib/valkyrie/specs/shared_specs/storage_adapter.rb

```diff
@@ -46,4 +50,26 @@ class CustomResource < Valkyrie::Resource
     expect { storage_adapter.find_by(id: uploaded_file.id) }.to raise_error Valkyrie::StorageAdapter::FileNotFound
     expect { storage_adapter.find_by(id: Valkyrie::ID.new("noexist")) }.to raise_error Valkyrie::StorageAdapter::FileNotFound
   end
+
+  it "can version files (if supported)" do
+    if storage_adapter.supports_versions?
+      resource = CustomResource.new(id: "test")
+      stored1 = storage_adapter.upload(file: file, original_filename: 'foo.jpg', resource: resource)
```

Comment:
I'm concerned about the interface here - it seems to just auto-version whenever you give it something with a similar original_filename and attach it to a resource. I've been thinking lately about an interface like one of these: https://gist.github.com/tpendragon/545f40274a2f55e106975cfe3d8ddde7

Reply:
To be clear original_filename is not used as part of a key in any way.  It's making a new version every time you say `upload`